source/server/listener_impl.cc

```diff
@@ -712,8 +712,7 @@ void ListenerImpl::buildSocketOptions() {
         break;
       }
       case envoy::config::listener::v3::Listener_ConnectionBalanceConfig::BALANCE_TYPE_NOT_SET: {
-        ENVOY_LOG(warn, "No valid balance type, switch to default NOP connection balancer");
-        break;
+        throw EnvoyException("No valid balance type for connection balance");
```

Comment:
This should have a test, thanks.

/wait

Reply:
In fact, if do not set any balance like:
```
static_resources:
  listeners:
  - address:
      socket_address:
        address: 0.0.0.0
        port_value: 9902
    connection_balance_config:
```

`config_.has_connection_balance_config()` will return `false`, since now we only have one value here, the BALANCE_TYPE_NOT_SET case will never run into, also can not add a test for it :(

Reply:
If set an invalid balance, it will raise an error when loading config.

Reply:
So I only add a test for invalid extend balance.

Reply:
Added, please review again, thanks.